ci-release.sh

```diff
@@ -20,10 +22,29 @@ fi
 git config user.name "CI release"
 git config user.email "ci-test@release.com"
 
+# Ensure the remote repository 'upstream' is added
+if ! git remote | grep -q 'upstream'; then
```

Comment:
How was the remote set previously?
Do we need to confirm the upstream target is correct as well?

Reply:
As we can see in the job [failure logs](https://storage.googleapis.com/test-platform-results/logs/branch-ci-openshift-backplane-cli-main-combined-release/1800401130698575872/build-log.txt): 
```
[36mINFO[0m[2024-06-11T05:34:23Z] Resolved source https://github.com/openshift/backplane-cli to main@6075bc81 
```
Looks the remote upstream was already resolved by the CI system but not pointing to a valid repo. I can update a bit to ensure repo is set to the correct url